terraform/environments/staging/main.tf

```diff
@@ -48,6 +48,8 @@ module "google-cloud-project" {
   name               = "Staging Environment"
   organization_id    = "335836213177"
   billing_account_id = "01DFC9-3D6951-579BE1"
+
+  auto_create_network = false
```

Comment:
This is a good catch of a diff between prod and staging.  I wonder if this is why you got the errors in staging in the first place when you made your changes.  Either way, it will be good to be as close to prod as possible so we can verify what will happen when we make the other changes to the Relay network setup.